Core/NosSmooth.Game/PacketHandlers/Map/CMapResponder.cs

```diff
@@ -84,6 +97,6 @@ await _game.CreateMapAsync
             ct: ct
         );
 
-        return Result.FromSuccess();
+        return await _eventDispatcher.DispatchEvent(new MapChangedEvent(previousMap, _game.CurrentMap), ct);
```

Comment:
Please check that the map is not null before dispatching the event. Return an error or throw an exception in case it's null - that should not happen. I've solved these with throwing UnreachableException in the past.
```cs
if (currentMap is not null)
{
  throw new UnreachableException();
}

return  await _eventDispatcher.DispatchEvent(new MapChangedEvent(previousMap, currentMap), ct);
```

Also be sure to use local variable currentMap instead of _game.CurrentMap as it's possible that _game.CurrentMap will change